### DISCUSSION TITLE
Customize schema names for embedded body params of auth endpoints

### DISCUSSION THREAD

**Discussion by bkis** [fastapi-users/fastapi-users#1213] [Q&A]
The schema names generated for the embedded body parameters of FastAPI-Users' endpoints are excessively verbose:

![Screenshot from 2023-05-11 15-11-46](https://github.com/fastapi-users/fastapi-users/assets/9215743/eada643c-ac16-4f88-8464-cf4a0a7e3592)  
![Screenshot from 2023-05-11 15-11-51](https://github.com/fastapi-users/fastapi-users/assets/9215743/e99e4b36-c7bf-4f00-ac9c-5a95b808054f)

This leads to a handful of very weird schema names:

![Screenshot from 2023-05-11 14-39-29](https://github.com/fastapi-users/fastapi-users/assets/9215743/110f1713-ee54-44c9-b163-e0f16bb4737a)

It's not just a cosmetic issue: I am generating TypeScript HTTP clients for my API using [OpenAPI Generator](https://openapi-generator.tech/) and of course these parameter names lead to **very** ugly client code:

![image](https://github.com/fastapi-users/fastapi-users/assets/9215743/2cb0d52b-ba12-407a-a9b0-d0bcd5f8f357)

Is there a way to do this? I am struggling to find a solution.

**Comment by frankie567** [comment]
The only way would be to refactor all routers to use properly defined Pydantic schemas instead of relying on inline `Body` dependencies.

Would be open to review a PR for this.

**Reply by bkis** [reply]
I'll look into it whenever I find some time to do so.

**Reply by bkis** [reply]
I put quite some time into this today. For the `request_verification_token` and `verify` endpoints, this was pretty straight forward: Add Pydantic models to match the request bodies and use those instead of the `Body(...)` dependencies. This just works, because the request type is `application/json` anyway.

But there are still the verbose names for any `login` routes' request body schemas. Those are `application/x-www-form-urlencoded` and I couldn't figure out how to make a Pydantic model work with the form data requirement. Actually, I did: I created a new class to replace FastAPI's `OAuth2PasswordRequestForm` using Pydantic's `BaseModel` and interfaced it with a classmethod that requires its arguments to be `Form()` fields. All tests pass, I was happy. But this didn't change those generated schema names :(

I am frustrated enough for today, maybe I'll try again another time.

**Reply by frankie567** [reply]
Thank you for looking into this!

Yes, unfortunately, I think with `Form(...)` dependencies, we won't be able to name the model, it's a limitation from FastAPI, AFAIK.
